src/main/java/school/hei/haapi/service/event/CorNotificationService.java

```diff
@@ -0,0 +1,112 @@
+package school.hei.haapi.service.event;
+
+import static school.hei.haapi.service.utils.DataFormatterUtils.formatLocalDateTime;
+import static school.hei.haapi.service.utils.InstantUtils.UTC3;
+import static school.hei.haapi.service.utils.TemplateUtils.htmlToString;
+
+import jakarta.mail.internet.AddressException;
+import jakarta.mail.internet.InternetAddress;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.thymeleaf.context.Context;
+import school.hei.haapi.endpoint.event.model.CorNotification;
+import school.hei.haapi.endpoint.rest.model.UserIdentifier;
+import school.hei.haapi.mail.Email;
+import school.hei.haapi.mail.Mailer;
+import school.hei.haapi.model.exception.BadRequestException;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class CorNotificationService implements Consumer<CorNotification> {
+  private Mailer mailer;
+
+  @Override
+  public void accept(CorNotification corNotification) {
+    var htmlBody = htmlToString("corNotification", getMailContext(corNotification));
+    mailer.accept(
+        new Email(
+            getDestinationEmail(corNotification),
+            List.of(),
+            getInterviewerEmails(corNotification),
+            "[COR] Convocation des parents",
+            htmlBody,
+            List.of()));
+  }
+
+  private Context getMailContext(CorNotification corNotification) {
+    var interviewDate = getInteviewLocalDateTime(corNotification);
+
+    var initial = new Context();
+    initial.setVariable("interviewers", corNotification.getCor().getInterviewers());
+    initial.setVariable("description", corNotification.getCor().getDescription());
+    initial.setVariable("date", formatLocalDateTime(interviewDate));
+    initial.setVariable("hour", toHourMinutes(interviewDate));
+    return initial;
+  }
+
+  private static LocalDateTime getInteviewLocalDateTime(CorNotification corNotification) {
+    var interviewInstant = corNotification.getCor().getInterviewDate();
+    if (interviewInstant == null) {
+      throw new BadRequestException(
+          "Interview date for the cor with id #%s is null"
+              .formatted(corNotification.getCor().getId()));
+    }
+    return LocalDateTime.ofInstant(interviewInstant, UTC3);
+  }
+
+  private static String toHourMinutes(LocalDateTime insertViewDate) {
+    return "%dh%d".formatted(insertViewDate.getHour(), insertViewDate.getMinute());
+  }
+
+  private static InternetAddress getDestinationEmail(CorNotification corNotification) {
+    var concernedStudent = corNotification.getCor().getConcernedStudent();
+    if (concernedStudent == null) {
+      throw new BadRequestException(
+          "Concerned student for the cor with id #%s is null"
+              .formatted(corNotification.getCor().getId()));
+    }
```

Comment:
nope, we are not in http context here, IllegalArgumentException
still isn't user required for Cor creation ?

```suggestion
	  log.error("Concerned student for the cor with id #%s is null", corNotification.getCor().getId());
	  return;
```

throwing error triggers a retry btw, Cor notification without user doesn't need to be retried isn't it?

Reply:
It must be sent to the dead letter letters, so that we can try again once the information is complete.